ONVAULT

```diff
@@ -43,28 +47,60 @@ if curl -s "${VAULT_URI}/_ping"; then
     cp -r ~/.ssh/* $tmp_ssh_vault
   fi
 
-  log "Downloading private keys..."
+  log "Downloading private keys and git credentials ..."
   curl -s "${VAULT_URI}/ssh.tgz" | tar -C ~/.ssh/ -zxf -
-  chown `whoami` ~/.ssh/*
-  chmod 600 ~/.ssh/*
+  chown -f `whoami` ~/.ssh/* || true
+  chmod -f 600 ~/.ssh/* || true
 
-  log "Using ssh key: $VAULT_SSH_KEY"
+  if [ ! -f ~/.ssh/$VAULT_SSH_KEY ]; then
+      log "Did not find ssh key: $VAULT_SSH_KEY"
+  else
+      log "Using ssh key: $VAULT_SSH_KEY"
+  fi
   if [[  "$VAULT_SSH_KEY" != "id_rsa" ]]; then
     # configure the ssh to any host to use this ssh key
     echo -e "\nHost *\nIdentityFile ~/.ssh/$VAULT_SSH_KEY" >> ~/.ssh/config
   fi
 
+  # download git credential store to temporary location
+  tmp_git_credential_store=`mktemp ~/.git-credentials-XXXXXX`
```

Comment:
I have removed the mktemp from the script to allow using it with Linux distros without this command.
Please take a look at c2029885994d1387532950388f1d1429beaa168d.